### PR TITLE
Make the specs swagger-friendly

### PR DESCRIPTION
**Pull request by jasebell**

### CODE REVIEW

File: src/kixi/spec.clj [hunk 1]
```diff
@@ -9,6 +10,35 @@
...
 
+(defmacro api-spec-uuid
+  [symb typename]
+  `(st/create-spec {:spec ~symb
+                    :form '~symb
+                    :json-schema/type ~typename
+                    :json-schema/format "uuid"}))
+
+
+(defmacro api-spec-array
```
Comment (otfrom) [inline src/kixi/spec.clj]: I don't understand why this is a macro

File: src/kixi/spec.clj [hunk 2]
```diff
@@ -9,6 +10,35 @@
                     :form '~symb
                     :json-schema/type ~typename}))
 
+(defmacro api-spec-uuid
```
Comment (otfrom) [inline src/kixi/spec.clj]: I don't understand why this is a macro

File: src/kixi/spec.clj [hunk 3]
```diff
@@ -9,6 +10,35 @@
...
+  [symb typename]
+  `(st/create-spec {:spec ~symb
+                    :form '~symb
+                    :json-schema/type "array"
+                    :json-schema/items {:type ~typename}}))
+
+(defn transform-keys [m]
+  (reduce-kv (fn [x y z] (assoc x (keyword "json-schema" (name y)) z)) {} m))
+
+(defmacro api-spec-explicit
```
Comment (otfrom) [inline src/kixi/spec.clj]: I don't understand why this is a macro

File: src/kixi/spec.clj (hunk 2 above)
Comment (acron0) [reply src/kixi/spec.clj]: This can be deleted

**Pull request merged**